packages/protoplugin/src/run-node.ts

```diff
@@ -46,8 +45,12 @@ export function runNodeJs(plugin: Plugin): void {
     .then((data) => {
       const req = CodeGeneratorRequest.fromBinary(data);
       const res = plugin.run(req);
-      process.stdout.write(res.toBinary());
-      process.exit(0);
+      process.stdout.write(res.toBinary(), (err) => {
+        if (err) {
+          process.exit(1);
+        }
+        process.exit(0);
+      });
     })
     .catch((reason) => {
```

Comment:
I think there is a potential issue with this. Node.js automatically quits when the script has executed. For example, consider this script:

```js
console.log("hello world");
```

ECMAScript promises are the cleanest solution to keep the event loop running for async operations. Consider:

```js
someOp().then(() => console.log("hello world"));
```

Node.js knows there's a pending promise, and won't quit before it resolves. Unfortunately, Node.js is older than Promises, and the entire network/stream API is based on event listeners and keeping the event loop running as long as listeners are attached.

The thing is, I'm not sure if the callback to stdout.write _consistently_ queues. Clearly it works in the cases we tested, but does it work on every platform/situation? I think we should err on the safe side, and use the promise chain we already maintain:

```suggestion
      return writeBytes(process.stdout, res.toBinary());
    })
    .then(() => process.exit(0))
    .catch((reason) => {
```

With the addition of:

```ts
/**
 * Write a chunk of bytes to a stream.
 */
function writeBytes(stream: WriteStream, data: Uint8Array): Promise<void> {
  return new Promise<void>((resolve, reject) => {
    stream.write(data, (err) => {
      if (err) {
        reject(err);
      } else {
        resolve();
      }
    });
  });
}
```

Reply:
Updated and verified locally.